Dev/GitHub/prnotii.5m.js

```diff
@@ -0,0 +1,108 @@
+  const body = [];
+  const req = https.get(url, {headers}, res => {
+    res.on('data', chunk => body.push(chunk));
+    res.on('end', () => resolve(Buffer.concat(body).toString()));
+  });
+  req.on('error', reject);
+  req.end();
+  setTimeout(reject, 5000, new Error('PRNOTII_ETIMEDOUT'));
+});
+const refine = items => items.map(item => ({
+  repository: item.repository_url.replace(/.+\//, ''),
+  href: item.html_url,
+  title: item.title,
+}));
+const organize = refined => refined.reduce((prev, item) => {
+  prev[item.repository] = prev[item.repository] || [];
+  prev[item.repository].push(item);
+  return prev;
+}, {});
+const githubQuery = (endpoint, token, query) => 
+  request(endpoint + '/search/issues?q=' + encodeURIComponent(query), {'Authorization': 'token ' + token})
+    .then(JSON.parse)
+    .then(res => refine(res.items))
+    .then(organize);
+const subMenu = organized => Object.entries(organized)
+  .map(([repository, items]) => '--' + repository + ' | disabled=true | size=12\n' + items.map(item => '--' + item.title + ' | href=' + item.href).join('\n'))
+  .join('\n');
+const printOut = str => str && console.log(str);
+
+(async function () {
+  try {
+    const configFile = join(homedir(), '.prnotii');
+    const {endpoint, token} = await readFile(configFile, {encoding: 'utf8'}).then(JSON.parse);
+
+    const requested = await githubQuery(endpoint, token, 'is:open is:pr user-review-requested:@me');
+    const requestedSubmenu = subMenu(requested);
+    printOut(`● | color=${Object.keys(requested).length ? 'red' : 'white'}`);
+    printOut('---');
+    printOut('👩‍💻 Review Requested');
+    printOut(requestedSubmenu);
+
+    const reviewed = await githubQuery(endpoint, token, 'is:pr is:open reviewed-by:@me');
+    const reviewedSubmenu = subMenu(reviewed);
+    printOut('💬 Reviewed');
+    printOut(reviewedSubmenu);
+
+    const created = await githubQuery(endpoint, token, 'is:open is:pr author:@me');
+    const createdSubmenu = subMenu(created);
+    printOut('💪🏻 Created');
+    printOut(createdSubmenu);
+
+    const mentioned = await githubQuery(endpoint, token, 'is:open is:pr mentions:@me');
+    const mentionedSubmenu = subMenu(mentioned);
+    printOut('😀 Mentioned');
+    printOut(mentionedSubmenu);
+  } catch (e) {
+    printOut(`● | color=yellow`);
+    printOut('---');
+
+    switch (true) {
+      case e.message.includes('ENOENT'):
+        printOut('🌈 HOW TO SETUP');
+        printOut('---');
+        printOut('❶ Generate Github Token');
+        printOut('⠀  Generate new personal access token with "repos scope". | size=12');
+        printOut('❷ Create .prnotii File');
+        printOut('⠀  Create ~/.prnotii File like below JSON format. | size=12');
+        printOut('⠀  {"token": "kSEzyDP...", "endpoint": "https://api.github.com"}  | size=12');
+        printOut('❸ Press Refresh button below');
+        printOut('⠀  After finish ❶,❷ steps, press "Refresh" button to start again. | size=12');
```

Comment:
This looks like a good use case for [xbar variables](https://xbarapp.com/docs/2021/03/14/variables-in-xbar.html). User would fill in the variables through the plugin browser UI, and the JSON configuration file is written automatically. You would just need to pull in the values via environment variables after that.

Not sure though if SwiftBar has a similar mechanism. If it doesn't, might be able to have the best of both systems with this flow:

1. Check for environment variables. If present, use them.
2. If no environment variables present, try reading the `.prnotii` configuration file.
3. If `.prnotii` not found, then display setup instructions. If able to detect we are running xbar versus swiftbar (I believe both have an env variable you could use for this purpose), write the setup instructions out for the appropriate program.

What are your thoughts?

Reply:
@sprak3000 
Thank you for the suggestion. I was able to learn about the xbar variables feature thanks to you. It seems like a perfect feature for this case. I have changed the implementation to handle the xbar variable file directly in order to minimize changes to the existing code.

Reply:
@eu81273 

I'm trying to test this locally and am getting `Unexpected token R in JSON at position 2`. I went through the xbar plugin browser UI to have it create the JSON configuration file. It looks like this:

```json
{
        "VAR_ENDPOINT": "https://api.github.com",
        "VAR_TOKEN": "ghp_..."
}
```

Any ideas on why the `JSON.parse` seems to be failing?

Reply:
The above was not the issue. Turns out not having a user agent in the request was. See latest review below.